### PR TITLE
Expire parameter renaming and deletion and attribute privatization from 3.5

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Will add a (quite long) release note.

Skipped widgets.py as it was a bit confusing at stages. But apart from that all renaming, deletion and privatization should be gone.

## PR Checklist

**Tests and Styling**
- [N/A] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/backends/backend_agg.py [hunk 1]
```diff
@@ -468,8 +467,7 @@ def _print_pil(self, filename_or_obj, fmt, pil_kwargs, metadata=None):
             filename_or_obj, self.buffer_rgba(), format=fmt, origin="upper",
             dpi=self.figure.dpi, metadata=metadata, pil_kwargs=pil_kwargs)
 
-    @_api.delete_parameter("3.5", "args")
-    def print_png(self, filename_or_obj, *args,
+    def print_png(self, filename_or_obj,
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_agg.py:470]: Once you expire a delete_parameter, all subsequent parameters also become kwonly (as it was impossible to pass them positionally during the deprecation period without triggering the deprecation). Ditto for all other insurances below.

Comment (oscargus) [reply lib/matplotlib/backends/backend_agg.py:470]: I was thinking about that. Makes sense!

**Comment by QuLogic** [comment]
For the removal notes, please add a bit more context than last time. When X is removed from Y where Y is 3 levels deep in an import tree, it is very hard to tell whether it's a relevant thing. See the changes in #23814 for examples.

**Comment by oscargus** [comment]
I've realized that the release notes is really the problem here. It may be that I close this and do parts gradually as for some of these I do not really have much context and the earlier notes does not give much info either.

**Comment by oscargus** [comment]
Closing this as it must be done in steps to be manageable.

**Pull request closed without merging**